src/PrestaShopBundle/Controller/Admin/Sell/CustomerService/MerchandiseReturnController.php

```diff
@@ -76,11 +91,198 @@ public function indexAction(Request $request, MerchandiseReturnFilters $filters)
+            $editableMerchandiseReturn = $this->getQueryBus()->handle(
+                new GetMerchandiseReturnForEditing(
+                    $merchandiseReturnId
+                )
+            );
+
+            $form = $formBuilder->getFormFor($merchandiseReturnId);
+            $form->handleRequest($request);
+
+            $result = $formHandler->handleFor($merchandiseReturnId, $form);
+
+            if ($result->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_merchandise_returns_index');
+            }
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        if (!isset($form)) {
+            return $this->redirectToRoute('admin_merchandise_returns_index');
+        }
+
+        return $this->render('@PrestaShop/Admin/Sell/CustomerService/MerchandiseReturn/edit.html.twig', [
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'enableSidebar' => true,
+            'layoutTitle' => sprintf($this->trans('Return Merchandise Authorization (RMA) ', 'Admin.Actions')),
+            'merchandiseReturnForm' => $form->createView(),
+            'editableMerchandiseReturn' => $editableMerchandiseReturn,
+            'merchandiseReturnsProductsGrid' => $this->presentGrid($gridFactory->getGrid($filters)),
+        ]);
+    }
+
+    /**
+     * @AdminSecurity("is_granted('delete', request.get('_legacy_controller'))", redirectRoute="admin_merchandise_returns_index")
+     *
+     * @param Request $request
+     * @param int $merchandiseReturnId
+     * @param int $merchandiseReturnDetailId
+     * @param int $customizationId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteProductAction(Request $request, int $merchandiseReturnId, int $merchandiseReturnDetailId, int $customizationId): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(
+                new DeleteProductFromMerchandiseReturnCommand($merchandiseReturnId, $merchandiseReturnDetailId, $customizationId)
+            );
+
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion.', 'Admin.Notifications.Success')
+            );
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $request->query->has('redirectUrl') ?
+            $this->redirect($request->query->get('redirectUrl')) :
+            $this->redirectToRoute(
+                'admin_merchandise_returns_edit',
+                [
+                    'merchandiseReturnId' => $merchandiseReturnId,
+                ]
+            );
+    }
+
+    /**
+     * I am not sure how to pass both via bulk
```

Comment:
It looks like you did it well 😊 or did I miss something ? Code looks fine here

Reply:
Yeah I already solved that, just forgot to remove this comment since didn't place @todo here